cmake/compiler/arcmwdt/generic.cmake

```
@@ -29,12 +29,22 @@ endif()
 execute_process(
   COMMAND ${CMAKE_C_COMPILER} --version
   RESULT_VARIABLE ret
-  OUTPUT_QUIET
+  OUTPUT_VARIABLE full_version_output
   ERROR_QUIET
   )
 
 if(ret)
   message(FATAL_ERROR "Executing the below command failed. Are permissions set correctly?
   '${CMAKE_C_COMPILER} --version'"
   )
+else()
+  string(REGEX MATCH "[A-Z]-[0-9][0-9][0-9][0-9]\.[0-9][0-9]"
+     vers_string ${full_version_output})
+
+  string(REGEX MATCH "[0-9][0-9][0-9][0-9]\.[0-9][0-9]"
+     reduced_version ${vers_string})
+
+  if( ${reduced_version} VERSION_LESS "2022.06")
```

please quote `${reduced_version}`, just in case the regex above results in an empty string.
```suggestion
  if("${reduced_version}" VERSION_LESS "2022.06")
```